airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -573,10 +575,17 @@ def execute_sync(self, context: Context):
                 self.pod, istio_enabled, self.base_container_name
             )
         finally:
-            self.cleanup(
-                pod=self.pod or self.pod_request_obj,
-                remote_pod=self.remote_pod,
-            )
+            try:
+                self.cleanup(
+                    pod=self.pod or self.pod_request_obj,
+                    remote_pod=self.remote_pod,
+                )
+            except Exception:
+                # If task got marked as failed, it should not raise exception (which might cause retry).
+                # https://github.com/apache/airflow/issues/36471
+                if not self._killed:
+                    raise
```

Comment:
I almost wonder if we should swallow the exception in `process_pod_deletion` instead, so we are a little more surgical with what we are ignoring.

Reply:
I think that probably won't fix this case as the exception was raised before `process_pod_deletion` is called

Reply:
Ah, good catch. I should have looked closer.  Now that I have though, should we even run cleanup after we hit on_kill? I don't see anything in there that we really need to run once we've decided to bail out?

Reply:
Indeed, just move the flag to `cleanup`. If the task has been killed, we'll not call it.